### DISCUSSION TITLE
PaymentSheet with StripeAccountId

### DISCUSSION THREAD

**Discussion by DevFreeze** [flutter-stripe/flutter_stripe#323] [Q&A] [answered]
Is it possible to pay with the PaymentSheet while using a StripeAccountId of a Connected Account?

As soon as I assign a StripeAccountId to `Stripe.stripeAccountId` the presentPaymentSheet function fails with the following error:

The client_secret provided does not match the client_secret associated with the PaymentIntent.

**Comment by remonh87** [comment]
The account id is something different then the client secret. When using the payment sheet you always need to add the client secret linked to the payment intent  in order to make it work. This client secret is created on the server. See also our examples

**Comment by DevFreeze** [comment]
Ok, so is there a way to use the PaymentSheet with Connected Accounts?

**Comment by DevFreeze** [comment, marked as answer]
OK, I managed to make a payment to an Connected Account with the PaymentSheet.

In addition to assigning the StripeAccountId to `Stripe.stripeAccountId` I need to also add it when the PaymentIntent is created:

```
exports.stripePayment = functions.https.onRequest(async (req, res) => {

    const amount = req.body.amount;
    const currency = req.body.currency;
    const fee = req.body.fee;
    const stripeAccountId = req.body.stripeAccountId;

    const paymentIntent = await stripe.paymentIntents.create({
        amount: amount,
        application_fee_amount: fee,
        currency: currency,
    }, {
        stripeAccount: stripeAccountId,
    });

    res.json({
        paymentIntent: paymentIntent.client_secret,
    });
});
```

**Reply by vytautas-pranskunas-** [reply]
1) Hmm thats strange because i have just talked to Stripe dev support and they said payments to connected accounts are possible only when you are creatig accounts in connected accounts which is not good practice. 

2) Also i do not see you creating 
```
 const ephemeralKey = await stripe.ephemeralKeys.create(
    { customer: customer.id },
    { apiVersion: '2020-08-27' }
  );
```

may I ask why?
3) Where do you assign customerId?

**Reply by vytautas-pranskunas-** [reply]
Would you be able to share full example please?
